src/TermInterface.jl

```diff
@@ -103,7 +103,7 @@ function similarterm(x, head, args, symtype = nothing; metadata = nothing)
   similarterm(typeof(x), head, args, symtype = symtype, metadata = metadata)
 end
 
-function similarterm(T::DataType, head, args, symtype = nothing; metadata = nothing)
+function similarterm(T::Type, head, args, symtype = nothing; metadata = nothing)
   head(args...)
 end
```

Comment:
I would error on fallback. This can be later implemented by providers of `T` that know that `head` is going to be a callable object. e.g. in SymbolicUtils `similarterm(T::Symbolic, head, args, ....) = head(args...)`

Reply:
In general I would make as less assumptions as possible on the types and structures of AST nodes other than they implement `istree`, `operation` and `arguments`

Reply:
At least in this package

Reply:
I'm okay with this if @shashi is fine with this

Reply:
Yes, I think error is good enough. But we need to see if this breaks CI.

Reply:
*downstream CI